src/app/test_executive/gossip_consistency.ml

```diff
@@ -0,0 +1,178 @@
+
+  (* TODO: find a way to avoid this type alias (first class module signatures restrictions make this tricky) *)
+  type network = Network.t
+
+  type node = Network.Node.t
+
+  type dsl = Dsl.t
+
+  let block_producer_balance = "1000" (* 1_000_000_000_000 *)
+
+  let config =
+    let n = 3 in
+    let open Test_config in
+    { default with
+      requires_graphql= true
+    ; block_producers=
+        List.init n ~f:(fun _ ->
+            {Block_producer.balance= block_producer_balance; timing= Untimed}
+        ) }
+
+  let expected_error_event_reprs = []
+
+  let pk_of_keypair keypairs n =
+    Public_key.compress (List.nth_exn keypairs n).Keypair.public_key
+
+  let wait_for_all_to_initialize ~logger network t =
+    let open Malleable_error.Let_syntax in
+    let producers = Network.block_producers network in
+    let n = List.length producers in
+    List.mapi producers ~f:(fun i node ->
+        let%map () = wait_for t (Wait_condition.node_to_initialize node) in
+        [%log info]
+          "gossip_consistency test: Block producer %d (of %d) initialized"
+          (i + 1) n ;
+        () )
+    |> Malleable_error.all_unit
+
+  let send_payments ~logger ~sender_pub_key ~receiver_pub_key ~amount ~fee
+      ~node n =
+    let open Malleable_error.Let_syntax in
+    let rec go n =
+      if n = 0 then return ()
+      else
+        let%bind () =
+          let%map () =
+            Network.Node.send_payment ~retry_on_graphql_error:false ~logger
+              ~sender_pub_key ~receiver_pub_key ~amount ~fee node
+          in
+          [%log info] "gossip_consistency test: payment #%d sent." n ;
+          ()
+        in
+        go (n - 1)
+    in
+    go n
+
+  let wait_for_payments ~logger ~dsl ~sender_pub_key ~receiver_pub_key ~amount
+      n =
+    let open Malleable_error.Let_syntax in
+    let rec go n =
+      if n = 0 then return ()
+      else
+        (* confirm payment *)
+        let%bind () =
+          let%map () =
+            wait_for dsl
+              (Wait_condition.payment_to_be_included_in_frontier
+                 ~sender_pub_key ~receiver_pub_key ~amount)
+          in
+          [%log info]
+            "gossip_consistency test: payment #%d successfully included in \
+             frontier."
+            n ;
+          ()
+        in
+        go (n - 1)
+    in
+    go n
+
+  let run network t =
+    let open Malleable_error.Let_syntax in
+    let logger = Logger.create () in
+    [%log info] "gossip_consistency test: starting..." ;
+    let%bind () = wait_for_all_to_initialize ~logger network t in
+    [%log info] "gossip_consistency test: done waiting for initializations" ;
+    let receiver_bp = Caml.List.nth (Network.block_producers network) 0 in
+    let%bind receiver_pub_key = Util.pub_key_of_node receiver_bp in
+    let sender_bp =
+      Core_kernel.List.nth_exn (Network.block_producers network) 1
+    in
+    let%bind sender_pub_key = Util.pub_key_of_node sender_bp in
+    let num_payments = 3 in
```

Comment:
@imeckler again, reduced num_payments this to iterate faster but can increase again